torch/_prims_common/__init__.py

```diff
@@ -924,24 +924,23 @@ def infer_size(shape: ShapeType, numel: int) -> tuple[int, ...]:
     Infers the size of a dim with size -1, if it exists.
     Also checks that new shape is compatible with the number of elements.
     """
+    from torch.fx.experimental.symbolic_shapes import definitely_true, guard_or_false
+
     dim = None
     newsize = 1
     for i, d in enumerate(shape):
-        if d == -1:
+        if guard_or_false(d == -1):
```

Comment:
guard_or_false(d == -1, error_message="We assumed d to be not -1 because it was unbacked if its not the case please add a runtime assert"):

Reply:
ah you have it sounds good